src/meshapi/views/lookups.py

```diff
@@ -1,12 +1,61 @@
 from django.db.models import Q
+from django_filters import rest_framework as filters
 from drf_spectacular.types import OpenApiTypes
 from drf_spectacular.utils import OpenApiParameter, extend_schema, extend_schema_view
-from rest_framework import filters, generics
+from rest_framework import generics
+from rest_framework.response import Response
 
-from meshapi.models import Building, Install, Member
-from meshapi.serializers import BuildingSerializer, InstallSerializer, MemberSerializer
+from meshapi.models import Building, Device, Install, Link, Member, Node, Sector
+from meshapi.serializers import (
+    BuildingSerializer,
+    DeviceSerializer,
+    InstallSerializer,
+    LinkSerializer,
+    MemberSerializer,
+    NodeSerializer,
+    SectorSerializer,
+)
+
+
+class FilterRequiredListAPIView(generics.ListAPIView):
+    def get(self, request, *args, **kwargs):
+        possible_filters = set(self.filterset_class.base_filters.keys())
+        provided_filters = set(self.request.query_params.keys())
+        invalid_filters = provided_filters - possible_filters
+
+        # If they gave us filters that aren't actually available, bail and return a bad status
+        if invalid_filters:
+            return Response({"detail": f"Invalid filters provided: {list(invalid_filters)}"}, 400)
+        else:
+            return super().get(request, *args, **kwargs)
+
+    def get_queryset(self):
+        result = super().get_queryset()
+
+        # This is a search endpoint, if the user doesn't supply any filters,
+        # they expect that we return nothing instead of everything.
+        # If they wanted everything they'd use the appropriate list endpoint
+        if not self.request.query_params:
+            return result.none()
```

Comment:
Maybe we should return a 400 instead of nothing?

Reply:
Done